src/components/appSwitch/AppSwitch.js

```diff
@@ -241,24 +238,31 @@ const AppSwitch = (props: LoadingProps) => {
 
   useEffect(() => {
     if (isMobileNative && DeepLinking.pathname) {
-      return props.navigation.navigate(DeepLinking.pathname.slice(1))
+      deepLinkingNavigation()
     }
   }, [])
 
   useEffect(() => {
     if (isMobileNative && DeepLinking.pathname) {
       DeepLinking.subscribe(() => {
-        return props.navigation.navigate(DeepLinking.pathname.slice(1))
+        deepLinkingNavigation()
       })
     }
+    return () => {
+      DeepLinking.unsubscribe()
+    }
   }, [DeepLinking.pathname])
 
   useEffect(() => {
-    AppState.addEventListener('change', handleAppFocus)
-    return function() {
-      AppState.removeEventListener('change', handleAppFocus)
+    if (ready && gdstore && appState === 'active') {
+      checkBonusInterval(true)
+      showOutOfGasError(props)
+      if (isMobileNative && !initSubscribe.current) {
```

Comment:
this is not necessary as you've already subscribed in the previous hook

Reply:
This implementation is a solution to a bug from react-native-branch. When the user opens the link and the app is open the listener does not work. Because of this, it tries to subscribe again.

Reply:
Hmm. let's try to incapsulate subscription logic in the single hook:

```javascript
useEffect(() => {
  if (!isMobileNative || appState !== 'active' || !DeepLinking.pathname) return
  
  DeepLinking.subscribe(deepLinkingNavigation)
  return () => DeepLinking.unsubscribe()
}, [DeepLinking.pathname, appState])
```